src/operation/mod.rs

```diff
@@ -103,6 +108,13 @@ pub(crate) trait Operation {
     fn retryability(&self) -> Retryability {
         Retryability::None
     }
+
+    // Updates this operation's write concern.
+    fn update_write_concern(&mut self) {}
```

Comment:
I think this is probably a bit too niche to include on the `Operation` trait, since I think it's unlikely other operations would need to use it. What about if we introduced a `fn build_retry` instead that `CommitTransaction` could implement which uses the appropriate write concern, and as a default implementation it just calls through to the regular `build`? This is a little more general and potentially could be useful in the future.

Reply:
Hm, that would require some refactoring since `build` is called further down in `execute_operation_on_connection` (which doesn't have any indication of whether we're on the first try or retry). Maybe we just rename this `update_for_retry` or something along those lines? I agree it's a little niche as-is.